signatures/golang/illegitimate_shell.go

```diff
@@ -54,16 +53,15 @@ func (sig *IllegitimateShell) OnEvent(event protocol.Event) error {
 	}
 
 	switch eventObj.EventName {
-	case "security_bprm_check":
+	case "sched_process_exec":
+		prevComm, err := helpers.GetTraceeStringArgumentByName(eventObj, "prev_comm")
+		if err != nil {
+			return err
+		}
 		for _, webServersProcessName := range sig.webServersProcessNames {
-			if webServersProcessName == eventObj.ProcessName {
-				pathname, err := helpers.GetTraceeStringArgumentByName(eventObj, "pathname")
-				if err != nil {
-					return err
-				}
-
+			if webServersProcessName == prevComm {
 				for _, shellName := range sig.shellNames {
-					if strings.HasSuffix(pathname, shellName) {
```

Comment:
I'm not sure but can't processName be controlled by the user, whereas pathName can't?

Reply:
pathname can be controlled by the user as well - altering the binary name, for example, will change the path suffix